corgi/core/models.py

```diff
@@ -1774,25 +1775,30 @@ def get_roots(self, using: str = "read_only") -> QuerySet[ComponentNode]:
 
     @property
     def cpes(self) -> QuerySet:
-        """Build and return a list of CPEs from all Variants this Component relates to"""
+        """Build and return a list of CPEs this Component relates to"""
         # For each Variant-type relation, get the linked Variant's CPE directly
         # Remove any duplicates, and return the CPEs in sorted order so manifests are stable
-        return (
-            self.productvariants.exclude(cpe="")
-            .values_list("cpe", flat=True)
-            .distinct()
-            .order_by("cpe")
-        )
-
-        # For Stream-type relations, we no longer link any Variants
-        # since this caused data quality issues, where Components were linked to Variants
-        # that didn't actually ship those Components
-        # We could look at all the linked streams to get CPEs from all their child Variants
-        # But this would cause the same problem - we'd report too many Variants / incorrect CPEs
+        if self.productvariants.exists():
+            return (
+                self.productvariants.exclude(cpe="")
+                .values_list("cpe", flat=True)
+                .distinct()
+                .order_by("cpe")
+            )
+        elif self.software_build:
+            build_variants = (
+                ProductComponentRelation.objects.filter(software_build=self.software_build)
+                .values_list("product_ref", flat=True)
+                .distinct()
```

Comment:
This is finding all relations, but I think we should filter only for relations that refer to variants:
```suggestion
                ProductComponentRelation.objects.filter(software_build=self.software_build, type__in=ProductComponentRelation.VARIANT_TYPES)
                .values_list("product_ref", flat=True)
                .distinct()
```
This also requires that the relation be processed. That's fine if it's what you want. But we could also report CPEs even for Variant product_refs from unprocessed relations. We'd just need to filter the relations on `build_id=self.software_build.build_id, build_type=self.software_build.build_type` instead of using `software_build=self.software_build` directly.

Reply:
Good idea to also filter by type. Although I think filtering by ERRATA type is probably best since that is the best indicator that the build has been released to that Variant. Errata type relations shouldn't ever not have a linked SoftwareBuild foreign key, so leaving that part of the filter unchanged.